test/connectivity/modules/NetworkController/FirewallOfflineController.cs

```diff
@@ -11,58 +10,33 @@ namespace NetworkController
 
-        public FirewallOfflineController(string networkInterfaceName)
+        public FirewallOfflineController(string networkInterfaceName, string iothubHosname)
         {
-            Preconditions.CheckNonWhiteSpace(networkInterfaceName, nameof(networkInterfaceName));
-
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                this.networkCommands = new WindowsFirewallCommands(networkInterfaceName);
+                this.underlyingController = new WindowsFirewallOfflineController(networkInterfaceName);
             }
             else
             {
-                this.networkCommands = new LinuxFirewallCommands(networkInterfaceName);
+                this.underlyingController = new LinuxFirewallOfflineController(networkInterfaceName, iothubHosname);
             }
         }
 
         public string Description => "FirewallOffline";
 
-        public async Task<NetworkStatus> GetStatus(CancellationToken cs) => await this.networkCommands.GetStatus(cs);
-
-        public Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
-        {
-            switch (status)
-            {
-                case NetworkStatus.Restricted:
-                    return this.AddNetworkControllingRule(cs);
-                case NetworkStatus.Default:
-                    return this.RemoveNetworkControllingRule(cs);
-                default:
-                    Log.LogDebug($"Status not set {status}");
-                    throw new NotSupportedException($"Status is not supported {status}");
-            }
-        }
-
-        async Task<bool> AddNetworkControllingRule(CancellationToken cs)
-        {
-            bool result = await this.networkCommands.AddDropRule(cs);
-
-            NetworkStatus status = await this.GetStatus(cs);
-            Log.LogInformation($"Command AddDropRule execution success {result}, network status {status}");
-
-            return result && status == NetworkStatus.Restricted;
-        }
+        public Task<NetworkStatus> GetStatus(CancellationToken cs) => this.underlyingController.GetStatus(cs);
 
-        async Task<bool> RemoveNetworkControllingRule(CancellationToken cs)
+        public async Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
         {
-            bool result = await this.networkCommands.RemoveDropRule(cs);
+            bool result = await this.underlyingController.SetStatus(status, cs);
 
-            NetworkStatus status = await this.GetStatus(cs);
-            Log.LogInformation($"Command RemoveDropRule execution success {result}, network status {status}");
+            string resultMessage = result ? "succeded" : "failed";
+            Log.LogInformation($"Command SetStatus {status} execution {resultMessage}, network status {status}");
 
-            return result && status == NetworkStatus.Default;
+            NetworkStatus reportedStatus = await this.GetStatus(cs);
+            return result && reportedStatus == status;
```

Comment:
if result is false, you don't need to run line38; just return false.

Reply:
I moved this code to Program, it does a get to log the current status

---
In reply to: [359638676](https://github.com/Azure/iotedge/pull/2137#discussion_r359638676) [](ancestors = 359638676)